src/repository/TreeRepository.ts

```diff
@@ -261,14 +266,18 @@ export class TreeRepository<Entity> extends Repository<Entity> {
         });
     }
 
-    protected buildChildrenEntityTree(entity: any, entities: any[], relationMaps: { id: any, parentId: any }[]): void {
+    protected buildChildrenEntityTree(entity: any, entities: any[], relationMaps: { id: any, parentId: any }[], options: (FindTreesOptions & { depth: number })): void {
```

Comment:
@tiagojsag @pleerock this definition requires me to add `depth` property to the `options` argument. 
 Since `FindTreesOptions` only has `depth` as optional property, is there a reason why `options` can't be optional and just use `FindTreesOptions` as it's type and not cause a breaking change?   As it is right now, it looks like I'll have to update `buildChildrenEntityTree` with `{ depth: -1 }` everywhere I don't want changing functionality.

Reply:
I think you may be correct, and this can probably be modified to be optional.